### PR TITLE
DO_NOT_MERGE: Ongoing effort of refactoring gradlew to fix code quality issues

### PR DESCRIPTION
**Pull request by Kreyren**
### Request
Would you be willing to provide an abstract to why are the components in that file written this way so that i could make a documentation and refactor accordingly? See comments

### CODE REVIEW

File: gradlew-next [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+# Need this for relative symlinks.
+while [ -h "$PRG" ] ; do
+    ls=`ls -ld "$PRG"`
+    link=`expr "$ls" : '.*-> \(.*\)$'`
+    if expr "$link" : '/.*' > /dev/null; then
+        PRG="$link"
+    else
+        PRG=`dirname "$PRG"`"/$link"
+    fi
+done
```
Comment (Kreyren) [inline gradlew-next]: Why do we need to set APP_HOME this way?

File: gradlew-next [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    ;;
+  MINGW* )
+    msys=true
+    ;;
+  NONSTOP* )
+    nonstop=true
+    ;;
+esac
+
+CLASSPATH=$APP_HOME/gradle/wrapper/gradle-wrapper.jar
```
Comment (Kreyren) [inline gradlew-next]: context?

File: gradlew-next [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+Please set the JAVA_HOME variable in your environment to match the
+location of your Java installation."
+    fi
+else
+    JAVACMD="java"
+    which java >/dev/null 2>&1 || die "ERROR: JAVA_HOME is not set and no 'java' command could be found in your PATH.
+
+Please set the JAVA_HOME variable in your environment to match the
+location of your Java installation."
+fi
```
Comment (Kreyren) [inline gradlew-next]: Are there any requirements for java command?

File: gradlew-next [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        fi
+    else
+        warn "Could not query maximum file descriptor limit: $MAX_FD_LIMIT"
+    fi
+fi
+
+# For Darwin, add options to specify how the application appears in the dock
+if $darwin; then
+    GRADLE_OPTS="$GRADLE_OPTS \"-Xdock:name=$APP_NAME\" \"-Xdock:icon=$APP_HOME/media/gradle.icns\""
+fi
```
Comment (Kreyren) [inline gradlew-next]: More info to darwin compatibility?

File: gradlew-next [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+        CHECK=`echo "$arg"|egrep -c "$OURCYGPATTERN" -`
+        CHECK2=`echo "$arg"|egrep -c "^-"`                                 ### Determine if an option
+
+        if [ $CHECK -ne 0 ] && [ $CHECK2 -eq 0 ] ; then                    ### Added a condition
+            eval `echo args$i`=`cygpath --path --ignore --mixed "$arg"`
+        else
+            eval `echo args$i`="\"$arg\""
+        fi
+        i=`expr $i + 1`
+    done
```
Comment (Kreyren) [inline gradlew-next]: More info to cygwin compatibility?

File: gradlew-next [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+save () {
+    for i do printf %s\\n "$i" | sed "s/'/'\\\\''/g;1s/^/'/;\$s/\$/' \\\\/" ; done
+    echo " "
+}
+APP_ARGS=`save "$@"`
+
+# Collect all arguments for the java command, following the shell quoting and substitution rules
+eval set -- $DEFAULT_JVM_OPTS $JAVA_OPTS $GRADLE_OPTS "\"-Dorg.gradle.appname=$APP_BASE_NAME\"" -classpath "\"$CLASSPATH\"" org.gradle.wrapper.GradleWrapperMain "$APP_ARGS"
+
+exec "$JAVACMD" "$@"
```
Comment (Kreyren) [inline gradlew-next]: context?

**Comment by big-guy** [comment]
Hey @Kreyren - you might get a better response on our Slack or forums.   Some of your questions may be best answered by changing the script with what you think is the better solution and seeing what works/breaks.  Thanks!

**Pull request closed without merging**